internal/distribution/distribution.go

```diff
@@ -0,0 +1,161 @@
+package distribution
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/labstack/echo/v4"
+)
+
+var DistributionNotFound = errors.New("Distribution not available")
+
+type DistributionItem struct {
+	Description string `json:"description"`
+	Name        string `json:"name"`
+}
+
+type Distributions []DistributionItem
+
+type DistributionFile struct {
+	ModulePlatformID string           `json:"module_platform_id"`
+	Distribution     DistributionItem `json:"distribution"`
+	ArchX86          *X86_64          `json:"x86_64,omitempty"`
+}
+
+type X86_64 struct {
```

Comment:
super-"nice to have" nitpick: Sorry that I missed this initially. This structure could have been named e.g. `Architecture`, because I expect that it could be reused as is, for any additional architectures supported by image-builder.

So now the `DistributionFile` would look like 
```go
type DistributionFile struct {
	ModulePlatformID string           `json:"module_platform_id"`
	Distribution     DistributionItem `json:"distribution"`
	ArchX86          *Architecture    `json:"x86_64,omitempty"`
}
```
and in the future could look like
```go
type DistributionFile struct {
	ModulePlatformID string           `json:"module_platform_id"`
	Distribution     DistributionItem `json:"distribution"`
	ArchX86          *Architecture    `json:"x86_64,omitempty"`
        ArchAarch64      *Architecture    `json:"aarch64,omitempty"`
}
```

Not a big deal, I just wanted to point this out.

Reply:
https://github.com/osbuild/image-builder/issues/339

Reply:
🙇